pyquil/quilbase.py

```diff
@@ -430,6 +430,12 @@ def _update_super(self, gate: quil_rs.Gate) -> None:
     def __str__(self) -> str:
         return super().to_quil_or_debug()
 
+    def __copy__(self) -> Self:
+        return self
```

Comment:
Could this just be implemented on `AbstractInstruction` then? instead of every subclass

Reply:
Ah, I remember now, we can't do that because these instruction classes inherit from both `AbstractInstruction` and a `quil-py` class. If both implement `__copy__`, then the base classes need to override it to resolve which one should be used.